tests/jballerina-unit-test/src/test/resources/test-src/services/client_resource_call_negative.bal

```diff
@@ -0,0 +1,62 @@
+type CustomString string;
+type CustomString2 CustomString;
+
+string? stringNilArgument = ();
+() nilArgument = ();
+
+CustomString customString = "x";
+CustomString2 customString2 = customString;
+
+const int intConstant = 1;
+var varInt = 1;
+var varBoolean = true;
+
+public function testClientResourceFunctionCallError() {
+    var successClient = client object {
+        resource function get path() returns int {
+            return 1;
+        }
+
+        resource function post path() returns int {
+            return 1;
+        }
+
+        resource function get path/[int a]() returns int {
+            return 1;
+        }
+
+        resource function get path/[int a]/foo(string name) returns int {
+            return 1;
+        }
+
+        resource function get path/[int a]/foo2(string name, string address) returns int {
+            return 1;
+        }
+    };
+
+    int result = successClient->/path/[10*11]/foo(123);
+    result = successClient->/path/[1]/foo("x", "y"); // No Error occurred
```

Comment:
@rdulmina, why is not this an error?

Reply:
Created a issue for this. https://github.com/ballerina-platform/ballerina-lang/issues/36687

Reply:
Fixed with [c6b8d7b](https://github.com/ballerina-platform/ballerina-lang/pull/36600/commits/c6b8d7beea75ac61238627c7b624be6cdaca6a40)